src/components/Home/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { Upload, Button, Icon, message } from 'antd';
+import api from '../../api';
+
+import { Container } from './styles';
+
+function Home() {
+  const [fileList, setFileList] = useState([]);
+  const [uploading, setUploading] = useState(false);
+
+  const handleUpload = () => {
+    const formData = new FormData();
+    fileList.forEach(file => {
+      formData.append('files[]', file);
+    });
+
+    setUploading(true);
+
+    // API Request
+    api
+      .post('/', formData)
+      .then(res => {
+        setFileList([]);
+        setUploading(false);
+        message.success('upload successfully.');
+      })
+      .catch(() => {
+        setUploading(false);
+        message.error('upload failed.');
+      });
+  };
+
+  const onRemoveFile = file => {
+    const index = fileList.indexOf(file);
+    const newFileList = fileList.slice();
+    newFileList.splice(index, 1);
```

Comment:
Eu vi uma galera reclamando, que se você manda um index inválido, no caso se não encontrar o index do arquivo né, o splice acaba removendo o último se não me engano.

Não sei se é real isso. Posso testar depois.

Mas se for real, daí seria bom colocar uma validação antes.

Reply:
Tem aquele recurso de deconstruct do ECMA6, funciona pra remoção também. Achei esse exemplo:

```js
const remove = (items,index) => {
  return [...items.slice(0,index),
          ...items.slice(index+1,items.length)];
};

const list = [1,2,3,4,5];
remove(list, 2);
// [1,2,3,4]
list
// [1,2,3,4,5]
```

Reply:
Sim! O ideal é ter uma validação sim ! Mas nesse caso ai, acho muito difícil ser inválido porque quem controla é o componente. Mas vale a observação